test/blackbox-tests/test-cases/pkg/opam-repository-download.t

```diff
@@ -161,3 +161,34 @@ restored the repo to where it was before)
   Solution for dune.lock:
   - bar.1.0.0
   - foo.0.1.0
+
+We also want to make sure that branches work, so we add `bar.2.0.0` as a new
+package on a `bar-2` branch (and switch back to the default branch, to make
+sure that the default branch differs from `bar-2`).
+
+  $ mkpkg bar 2.0.0 <<EOF
+  > depends: [ "foo" ]
+  > EOF
+  $ cd mock-opam-repository
+  $ git switch --quiet -c bar-2
```

Comment:
This test depends on git, right?

Reply:
Yes, we ended up making the `:whole_subtree` depend on `git` some time ago.